profile/src/v100/entity/account/account.rs

```diff
@@ -0,0 +1,687 @@
+    Hash,
+    Eq,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[display("{display_name} | {address}")]
+#[serde(rename_all = "camelCase")]
+pub struct Account {
+    /// The ID of the network this account can be used with.
+    #[serde(rename = "networkID")]
+    pub network_id: NetworkID,
+
+    /// A globally unique identifier of this account, being a human readable
+    /// address of an account. Always starts with `"account_"``, for example:
+    ///
+    /// `account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease`
+    ///
+    /// Most commonly the user will see this address in its abbreviated
+    /// form which is:
+    ///
+    /// `acco...please`
+    ///
+    /// No two addresses will ever be the same even for the same factor source
+    /// but on different networks, since the public keys controlling the
+    /// accounts depend on the network id.
+    pub address: AccountAddress,
+
+    /// An off-ledger display name or description chosen by the user when she
+    /// created this account.
+    pub display_name: DisplayName,
+
+    /// Security state of this account, either "securified" or not.
+    pub security_state: EntitySecurityState,
+
+    /// The visual cue user learns to associated this account with, typically
+    /// a beautiful colorful gradient.
+    #[serde(rename = "appearanceID")]
+    pub appearance_id: AppearanceID,
+
+    /// An order set of `EntityFlag`s used to describe certain Off-ledger
+    /// user state about Accounts or Personas, such as if an entity is
+    /// marked as hidden or not.
+    #[serde(default)]
+    pub flags: EntityFlags,
+
+    /// The on ledger synced settings for this account, contains e.g.
+    /// ThirdPartyDeposit settings, with deposit rules for assets.
+    pub on_ledger_settings: OnLedgerSettings,
+}
+
+impl Account {
+    pub fn new(
+        account_creating_factor_instance: HDFactorInstanceAccountCreation,
+        display_name: DisplayName,
+        appearance_id: AppearanceID,
+    ) -> Self {
+        let address =
+            AccountAddress::from_hd_factor_instance_virtual_entity_creation(
+                account_creating_factor_instance.clone(),
+            );
+        Self {
+            network_id: account_creating_factor_instance.network_id(),
+            address,
+            display_name,
+            security_state:
+                UnsecuredEntityControl::with_entity_creating_factor_instance(
+                    account_creating_factor_instance,
+                )
+                .into(),
+            appearance_id,
+            flags: EntityFlags::default(),
+            on_ledger_settings: OnLedgerSettings::default(),
+        }
+    }
+}
+
+impl Identifiable for Account {
+    type ID = AccountAddress;
+
+    fn id(&self) -> Self::ID {
+        self.address.clone()
+    }
+}
+
+impl PartialOrd for Account {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Account {
```

Comment:
FIXME: try derive it `Ord` instead, maybe all the types in field now impl `Ord` (did not before)